napari/__main__.py

```diff
@@ -162,14 +163,26 @@ def main():
         # so remove --plugin from sys.argv to prevent that warningz
         sys.argv.remove('--plugin')
 
-    with gui_qt(startup_logo=True):
-        view_path(
-            args.images,
-            stack=args.stack,
-            plugin=args.plugin,
-            layer_type=args.layer_type,
-            **kwargs,
-        )
+    if any(p.endswith('.py') for p in args.paths):
+        if len(args.paths) > 1:
+            sys.exit(
+                'When providing a python script, only a '
+                'single positional argument may be provided'
+            )
+
+        with gui_qt(startup_logo=False) as app:
```

Comment:
I'd go `startup_logo=True` for launching here, especially as file may do some complex loading, will be nice to give feedback right away that something is happening

Reply:
problem is that the start logo then doesn't go away.  If you want that behavior, I can add the `splash_widget` as an attribute of the application in `event_loop.py` and then manually close it in the CLI?

Reply:
just applied that hack.  try again

Reply:
Hmm it seemed to work for me already - see this gif, where I did `startup_logo=True` and ran `add_image.py` without the context manager there

![launch](https://user-images.githubusercontent.com/6531703/85229356-1662cd80-b39e-11ea-8a6b-2be9365cc792.gif)

Interestingly though if the `.py` file also contains a context manager then it just hangs. Is that expected?

I will try your new approach now

Reply:
Now it with your new approach it works regardless, and it's actually nicer as the splash screen closes faster and doesn't overlap with the viewer. I like this behaviour quite a lot:

![launch_2](https://user-images.githubusercontent.com/6531703/85229460-a7d23f80-b39e-11ea-88da-fb470b3cba21.gif)

Reply:
yeah, the previous one didn't work, but only if you had a context manager in the script.  that was the problem